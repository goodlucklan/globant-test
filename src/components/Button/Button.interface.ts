import { TouchableOpacityProps } from "react-native";

export interface ButtonProps extends Omit<TouchableOpacityProps, "title"> {
  isLoading?: boolean;
}
