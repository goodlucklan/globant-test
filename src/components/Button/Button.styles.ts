import { StyleSheet } from "react-native";
import { pallete } from "../../styles/variables";

export const styles = StyleSheet.create({
  button: {
    alignItems: "center",
    justifyContent: "center",
    borderRadius: 4,
    backgroundColor: pallete.black,
    paddingVertical: 8,
    paddingHorizontal: 24,
    flexDirection: "row",
  },
  textButton: {
    color: pallete.white,
    textAlign: "center",
  },
});
