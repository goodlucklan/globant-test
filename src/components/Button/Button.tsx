import { View, Text, TouchableOpacity, ActivityIndicator } from "react-native";
import React from "react";
import { ButtonProps as Props } from "./Button.interface";
import { styles } from "./Button.styles";
import { pallete } from "../../styles/variables";
const Button: React.FC<Props> = (props) => {
  const { onPress, isLoading = false } = props;
  return (
    <TouchableOpacity style={styles.button} onPress={onPress}>
      {isLoading ? <ActivityIndicator color={pallete.green} /> : null}
      <Text style={styles.textButton}>{props.children}</Text>
    </TouchableOpacity>
  );
};

export default Button;
