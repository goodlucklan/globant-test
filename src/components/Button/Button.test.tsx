import Button from "./Button";

describe("Button Component", () => {
  test("render a global Component", () => {
    expect(<Button />).toMatchSnapshot();
  });
});
