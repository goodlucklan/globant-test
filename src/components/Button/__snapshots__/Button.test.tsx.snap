// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component render a global Component 1`] = `<Button />`;
